src/components/UploaderForm.test.tsx

```diff
@@ -0,0 +1,22 @@
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+import UploaderForm from './UploaderForm';
+import { InternalApiGateway } from '../gateways/internal-api';
+
+test('it renders an uploader panel and a continue button', async () => {
+  const gateway = new InternalApiGateway();
+  const documentTypes = await gateway.getDocumentTypes(true);
+  const evidenceRequest = await gateway.getEvidenceRequest(true);
+
+  render(
+    <UploaderForm
+      requestId="foo"
+      documentTypes={documentTypes}
+      evidenceRequest={evidenceRequest}
+    />
+  );
+  expect(screen.getByText('Passport'));
+  expect(screen.getByText('A valid passport open at the photo page'));
+  expect(screen.getByText('Browse...'));
+  expect(screen.getByText('Continue'));
+});
```

Comment:
i've added assertions to check for proper validation. i'm not sure what the right behaviour for submission is, so i can't write a test for it

Reply:
That's exactly the kind of question that should be prompted when designing a component and testing it. In this case, even though we don't know the submit behaviour in the app as a whole, can we design this component so that it is detached from that?

I.e. this component should probably accept a prop (something like `onSubmit`) which should be called with the files to be uploaded, that way we can test that and it puts it in the hands of the parent to decide what to do with it.

I would also advise that we move the `FormData` stuff out of this component, it has no business serialising the data for upload—that should be done somewhere else, this is just a view component to capture the data.

Reply:
if using formdata is necessary to capture the file data, it belongs here, no?

i've taken it out for now. none of the forms have proper handlers yet, so this one is like the others